at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -30,6 +33,9 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
   late final AtClient _atClient;
   late final RemoteSecondary _remoteSecondary;
   late final NotificationServiceImpl _statsNotificationListener;
+
+  /// static because once listeners are added, they should be agnostic to switch atsign event
+  static final Set<SyncProgressListener> _syncProgressListeners = HashSet();
```

Comment:
Why is this static? the listeners wish to subscribe to a specific sync service

What is expected behaviour of sync progress listeners when atsign is switched?

Reply:
@gkc I have requested that for the following reasons:

If we don't make it static we need:

1. Every switch @sign is followed by a call to SyncServiceImpl.addListener() too. Which needs some education for the app dev's.

2. Generally state is not specific to an instance of SyncServiceImpl

Please let me know your thoughts.

Reply:
e.g alice registered a sync progress listener..app switches to bob. Now alice's listener is retained due to static nature.Otherwise caller has to register the listener again or we have to handle to logic to remember the registered listeners per atsign, when atsign is switched.

Reply:
So the contract is that if I subscribe to sync progress events, I will get them for all atSigns - ok that makes sense (needs to be made clear)

How do I, the subscriber, know which atSign the event was for? Maybe we should add atSign into SyncProgressEvent ?

Reply:
I have made the changes.

Reply:
Thanks Murali, LGTM